airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/JdbcSourceOperations.java

```diff
@@ -317,9 +317,9 @@ public JsonSchemaPrimitive getType(final JDBCType bigQueryType) {
       case REAL -> JsonSchemaPrimitive.NUMBER;
       case NUMERIC, DECIMAL -> JsonSchemaPrimitive.NUMBER;
       case CHAR, NCHAR, NVARCHAR, VARCHAR, LONGVARCHAR -> JsonSchemaPrimitive.STRING;
-      case DATE -> JsonSchemaPrimitive.STRING;
-      case TIME -> JsonSchemaPrimitive.STRING;
-      case TIMESTAMP -> JsonSchemaPrimitive.STRING;
+      case DATE -> JsonSchemaPrimitive.STRING_DATE;
```

Comment:
I think we need to be super careful about how MySQL/Postgres/MSSQL/Oracle handle timezones. For example, from the MySQL docs: 

> MySQL converts TIMESTAMP values from the current time zone to UTC for storage, and back from UTC to the current time zone for retrieval. (This does not occur for other types such as DATETIME.) By default, the current time zone for each connection is the server's time. The time zone can be set on a per-connection basis. As long as the time zone setting remains constant, you get back the same value you store. If you store a TIMESTAMP value, and then change the time zone and retrieve the value, the retrieved value is different from the value you stored. This occurs because the same time zone was not used for conversion in both directions. The current time zone is available as the value of the time_zone system variable. For more information, see Section 5.1.15, “MySQL Server Time Zone Support”.

So for MySQL to correctly express timestamps we would need to set the connection timezone to UTC whenever we're reading the date types in UTC (we should also have tests for this)

wdyt?

Reply:
good point. But currently jdbc destinations can not handle different date-time datatypes for databases and just write value as string

Reply:
@VitaliiMaltsev 

> currently jdbc destinations can not handle different date-time datatypes for databases and just write value as string

I believe this is inaccurate - [normalization writes appropriately typed date/time types depending on the value in the `format` field](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py#L500)